tekton/src/pipelines/fvt-upgrade-pre.yml.j2

```diff
@@ -82,6 +85,46 @@ spec:
         - input: "$(params.fvt_digest_ctf)"
           operator: notin
           values: [""]
+        - input: "$(params.mas_app_channel_monitor)" # Only run if Monitor channel is declared
+          operator: notin
+          values: [""]
+      workspaces:
+        - name: configs
+          workspace: shared-configs
+    
+    # Optimizer Tests Pre Upgrade
+    - name: fvt-optimizer-upgrade-pre
+      params:
+        - name: mas_instance_id
+          value: $(params.mas_instance_id)
+        - name: mas_workspace_id
+          value: $(params.mas_workspace_id)
+        - name: fvt_image_registry
+          value: $(params.fvt_image_registry)
+        - name: fvt_image_digest
+          value: $(params.fvt_digest_ctf)
+        - name: fvt_test_suite
+          value: optimizer_upgrade_pre # Pre upgrade tests
+        - name: fvt_test_dir
+          value: optimizer # Test directory to lookup for tests. It will be relative to workingDir value (eg. /opt/ibm/test/src)
+        - name: product_channel
+          value: $(params.mas_app_channel_optimizer)
+        - name: fvt_image_namespace
+          value: ai-solutions
+        - name: fvt_image_name
+          value: common-test-framework
+        - name: product_id
+          value: ibm-mas-optimizer
+      taskRef:
+        kind: Task
+        name: mas-fvt-run-suite
+      when:
+        - input: "$(params.fvt_digest_ctf)"
+          operator: notin
+          values: [""]
+        - input: "$(params.mas_app_channel_optimizer)" # Only run if Optimizer channel is declared
```

Comment:
Hi @alequint. I wanted to check with you if this validation is fine. It was required to control when to run the pipeline based on whether the app was installed or not. Basically, I added a validation to check if the app channel was declared. If so, it means the product is installed and the suite should run against it. Without this control, the suite was failing for obvious reasons since the Monitor was not installed in the environment. With this control, we can run the proper suite based on the installed app.

Reply:
The same was made for Monitor [here](https://github.com/ibm-mas/cli/pull/1172/files#diff-2eb64eaf2b101d3d4a8d4c2f03dab9a9174de4ee0d00a52f123385bbdd74415eR94-R96)